microsoft/vscode
# prevents cursor movement (auto-search) while typing in Find Widget

Hi!  First, this is my first pull request on _any_ project, so please accept my apologies if I haven't followed protocol.  If so, I assure you it's through ignorance, not malice, and I'm glad to learn or revise.  End of pre-apology.  

This PR fixes #60977 [edit: and fixes #70306 , addressed in a separate comment below]
60977: Disable auto-searching while typing
70306: Editor jumps to random line when clicking into it

**Brief summary of the issue**

- When using the Find Widget, the cursor moves as you type, resulting in disorienting editor scrolling to off-screen "pre-matches" before you've actually finished typing what you wanted to search for.  
- In general, we do not wish to scroll to the first _letter_ of our `find` term; we want to only go to the *whole* match.  And even then, we may not wish to move at all, but rather just check to see whether what we are looking for is on-screen.  


**What did you do to fix the problem?** 

In logical terms: 

1. highlight on `input` into the widget, 
2. but scroll _only_ upon `enter` press (or `click`ing on the widget up/down arrows.  

In code terms:

1. deleted the `moveCursor` code block within `private research(moveCursor, newFindScope)`  
2. step (1) alone was sufficient to solve the problem without causing an problems, but 
3. removed the `moveCursor` parameter from `research` and cleaned up references so that `research` now has a single parameter: `newFindScope`.   

A more sophisticated approach would be to create a `settings.json` option to dis/enable cursor movement while typing.  However, given the idiosyncrasies of the cases below, I'm hard pressed to think of when they are desirable over the newly-fixed behavior, where cursor movement is always intentional, and just an enter-press (or click) away.  


**How to test for fixed behavior** 

0. First, observe the former, bugged behavior for comparison, either (a) below in test `find` scenario details element, or (b) in GIFs from #60977 
1. copy the code block below into the editor
2. follow the search scenario in the "Details" element below
3. note that the pathological cases no longer occur, and
4. desirable behaviors are maintained, namely: (a) highlight-as-you-type, (b) to jump to next match, press Enter or click widget up/down arrows (c) find-and-replace follows (a) and (b) as well.  

**test `find` scenario, and cases that illustrate prior, bugged behavior**

<details>  

use the `Find` Widget to look for `Z7` in code block in the `details` element below

- Case 1: `find` string is **not** on current screen --> editor scrolls to top on type (disorienting)
- Case 2: `find` string **is** on screen, **after** cursor --> editor highlights, no scroll (clear) 
- Case 3: `find` string **is** on screen, **before** cursor --> editor scrolls to top (disorienting)

This block is long so that it takes up more than one screen height.  If your screen is long enough to fit it all, you won't observe the behavior; in that case, just add more numerical blocks to fill it out.  

**As is, it represents Case 1.  To reproduce Cases 2 and 3, simply change one of the `7`s in the bottom block to a `Z`.**  


```
Z

0
1
2
3
4
5
6
7
8
9

00
11
22
33
44
55
66
77
88
99

000
111
222
333
444
555
666
777
888
999

0000
1111
2222
3333
4444
5555
6666
7777
8888
9999

00000
11111
22222
33333
44444
55555
66666
77777
88888
99999

000000
111111
222222
333333
444444
555555
666666
777777
888888
999999

0000000
1111111
2222222
3333333
4444444
5555555
6666666
7777777
8888888
9999999

00000000
11111111
22222222
33333333
44444444
55555555
66666666
77777777
88888888
99999999


000000000
111111111
222222222
333333333
444444444
555555555
666666666
777777777
888888888
999999999

0000000000
1111111111
2222222222
3333333333
4444444444
5555555555
6666666666
7777777777
8888888888
9999999999

00000000000
11111111111
22222222222
33333333333
44444444444
55555555555
66666666666
77777777777
88888888888
99999999999

000000000000
111111111111
222222222222
333333333333
444444444444
555555555555
666666666666
777777777777
888888888888
999999999999

0000000000000
1111111111111
2222222222222
3333333333333
4444444444444
5555555555555
6666666666666
7777777777777
8888888888888
9999999999999

00000000000000
11111111111111
22222222222222
33333333333333
44444444444444
55555555555555
66666666666666
77777777777777
88888888888888
99999999999999

000000000000000
111111111111111
222222222222222
333333333333333
444444444444444
555555555555555
666666666666666
777777777777777
888888888888888
999999999999999

0000000000000000
1111111111111111
2222222222222222
3333333333333333
4444444444444444
5555555555555555
6666666666666666
7777777777777777
8888888888888888
9999999999999999

00000000000000000
11111111111111111
22222222222222222
33333333333333333
44444444444444444
55555555555555555
66666666666666666
77777777777777777
88888888888888888
99999999999999999

1. `find` string is NOT on current screen --> editor scrolls to top on type (disorienting)   :(
2. `find` string IS on screen, AFTER cursor --> editor highlights, no scroll (clear)          :)
3. `find` string IS on screen, BEFORE cursor --> editor scrolls to top (counterproductive)  :(

000000000000000000
111111111111111111
222222222222222222
333333333333333333
444444444444444444
555555555555555555
666666666666666666
777777777777777777
888888888888888888
999999999999999999

```
  </details>

Sorry for writing too much :/

Happy to fix this another way, if I was too hack and slash.  But at least it worked!  :)
